opengrok-indexer/src/test/java/org/opengrok/indexer/history/PerforceRepositoryTest.java

```diff
@@ -105,4 +106,25 @@ public void testHistoryAndAnnotations() throws Exception {
             }
         }
     }
+
+    @Test
+    public void testProtectFilename() throws Exception {
+        PerforceRepository instance = new PerforceRepository();
+        ArrayList<SimpleImmutableEntry<String,String>>   testmap = new ArrayList<SimpleImmutableEntry<String,String>>() {
```

Comment:
I am pretty sure the diamond operator does not have to be fully specified on both sides.

Reply:
I could not get it to compile without the inner types, I tried numerous omissions.  I am not very good at java, so may be wrong, but I didn't see how to make it work.

Reply:
Oh, this cannot be done because you are using anonymous inner class for initializing the array list.

The alternative I had in mind:
```java
        ArrayList<SimpleImmutableEntry<String,String>> testmap = new ArrayList<>();
        testmap.add(new SimpleImmutableEntry<>("Testfile 34", "Testfile 34"));
        testmap.add(new SimpleImmutableEntry<>("Test%52", "Test%2552"));
...
```

Reply:
Ahh, okay.  I tried to be fancy by initializing in the declaration.  But, I see the value in the simpler approach you have above.  I'll do that, thanks.

Reply:
Related question.  I have an instance of the class instantiated here, but am only testing a static method.  Would it be preferable just to call the static method, or instantiate a class object in this test?

Reply:
Actually, before I push up my last commit.  @vladak - How would you prefer I engage this method in this test?  From an instanced instance of PerforceRepository, or just as a static method in PerforceRepository?  Neither is more correct, I don't think, so I leave it to preference.  Let me know, so I can take out the question comment here on that subject.

Reply:
Just use the static method directly.